src/citrine/resources/design_execution.py

```diff
@@ -156,24 +156,24 @@ def update(self, model: DesignExecution) -> DesignExecution:
         """Cannot update an execution."""
         raise NotImplementedError("Cannot update a DesignExecution.")
 
-    def archive(self, execution_id: UUID):
+    def archive(self, uid: Union[UUID, str]):
```

Comment:
For this and other methods that throw an error, might as well just do the breaking migration